includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -1710,26 +1717,38 @@ private function add_order_info_to_object( $charge_id, $object ) {
 		// If the order couldn't be retrieved, return an empty order.
 		$object['order'] = null;
 		if ( $order ) {
-			$object['order'] = [
-				'number'       => $order->get_order_number(),
-				'url'          => $order->get_edit_order_url(),
-				'customer_url' => $this->get_customer_url( $order ),
-			];
+			$object['order'] = $this->build_order_info( $order );
+		}
 
-			if ( function_exists( 'wcs_get_subscriptions_for_order' ) ) {
-				$object['order']['subscriptions'] = [];
+		return $object;
+	}
 
-				$subscriptions = wcs_get_subscriptions_for_order( $order, [ 'order_type' => [ 'parent', 'renewal' ] ] );
-				foreach ( $subscriptions as $subscription ) {
-					$object['order']['subscriptions'][] = [
-						'number' => $subscription->get_order_number(),
-						'url'    => $subscription->get_edit_order_url(),
-					];
-				}
+
+	/**
+	 * Creates the array representing order for frontend.
+	 *
+	 * @param WC_Order $order The order.
+	 * @return array
+	 */
+	private function build_order_info( $order ): array {
```

Comment:
Let's harden the method: `private function build_order_info( WC_Order $order ): array {`